modules/policy/src/main/java/org/eclipse/xpanse/modules/policy/policyman/PolicyManager.java

```diff
@@ -0,0 +1,309 @@
+            String errorMsg = String.format("The policy with id %s not found.",
+                    updateRequest.getId());
+            throw new PolicyNotFound(errorMsg);
+        }
+        if (!StringUtils.equals(getCurrentLoginUserId(), existedEntity.getUserId())) {
+            throw new AccessDeniedException(
+                    "No permissions to update policy belonging to other users.");
+        }
+        boolean updatePolicy = false;
+        if (StringUtils.isNotBlank(updateRequest.getPolicy())
+                && StringUtils.equals(updateRequest.getPolicy(), existedEntity.getPolicy())) {
+            existedEntity.setPolicy(updateRequest.getPolicy());
+            updatePolicy = true;
+        }
+
+        boolean updateCsp = false;
+        if (Objects.nonNull(updateRequest.getCsp())) {
+            existedEntity.setCsp(updateRequest.getCsp());
+            updateCsp = true;
+        }
+        if (Objects.nonNull(updateRequest.getEnabled())) {
+            existedEntity.setEnabled(updateRequest.getEnabled());
+        }
+        if (updatePolicy) {
+            validatePolicy(existedEntity.getPolicy());
+            checkPolicyWhetherCreated(existedEntity.getCsp(), existedEntity.getPolicy());
+        }
+        if (updateCsp) {
+            checkPolicyWhetherCreated(existedEntity.getCsp(), existedEntity.getPolicy());
+        }
+
+        PolicyEntity updatedPolicy = policyStorage.store(existedEntity);
+        return conventToPolicyVo(updatedPolicy);
+    }
+
+    /**
+     * Get details of the policy.
+     *
+     * @param id the id of the policy.
+     * @return Returns the policy view object.
+     */
+    public PolicyVo getPolicyDetails(UUID id) {
+        PolicyEntity existedEntity = policyStorage.findPolicyById(id);
+        if (Objects.isNull(existedEntity)) {
+            String errorMsg = String.format("The policy with id %s not found.", id);
+            throw new PolicyNotFound(errorMsg);
+        }
+        if (!StringUtils.equals(getCurrentLoginUserId(), existedEntity.getUserId())) {
+            throw new AccessDeniedException(
+                    "No permissions to view policy belonging to other users.");
+        }
+        return conventToPolicyVo(existedEntity);
+    }
+
+
+    /**
+     * Delete the policy.
+     *
+     * @param id the id of policy.
+     */
+    public void deletePolicy(UUID id) {
+        PolicyEntity existedEntity = policyStorage.findPolicyById(id);
+        if (Objects.isNull(existedEntity)) {
+            String errorMsg = String.format("The policy with id %s not found.", id);
+            throw new PolicyNotFound(errorMsg);
+        }
+        if (!StringUtils.equals(getCurrentLoginUserId(), existedEntity.getUserId())) {
+            throw new AccessDeniedException(
+                    "No permissions to delete policy belonging to other users.");
+        }
+        policyStorage.deletePolicyById(id);
+    }
+
+
+    private String getCurrentLoginUserId() {
```

Comment:
we have this method duplicated at many places. We must move it to one single common place.